File: weather-app/src/main/java/rcgonzalezf/org/weather/common/analytics/NetworkType.java
package rcgonzalezf.org.weather.common.analytics;

import android.support.annotation.StringDef;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

import static rcgonzalezf.org.weather.common.analytics.AnalyticsManager.MOBILE;
import static rcgonzalezf.org.weather.common.analytics.AnalyticsManager.NONE;
import static rcgonzalezf.org.weather.common.analytics.AnalyticsManager.UNKNOWN;
import static rcgonzalezf.org.weather.common.analytics.AnalyticsManager.WIFI;

@StringDef({ MOBILE, UNKNOWN, NONE, WIFI }) @Retention(RetentionPolicy.SOURCE) @interface NetworkType {
}

File: weather-app/src/main/java/rcgonzalezf/org/weather/common/analytics/NetworkType.kt
package rcgonzalezf.org.weather.common.analytics

import androidx.annotation.StringDef
import rcgonzalezf.org.weather.common.analytics.AnalyticsManager.Companion.MOBILE
import rcgonzalezf.org.weather.common.analytics.AnalyticsManager.Companion.NONE
import rcgonzalezf.org.weather.common.analytics.AnalyticsManager.Companion.UNKNOWN
import rcgonzalezf.org.weather.common.analytics.AnalyticsManager.Companion.WIFI

@StringDef(MOBILE, UNKNOWN, NONE, WIFI)
@kotlin.annotation.Retention(AnnotationRetention.SOURCE)
internal annotation class NetworkType
